Modules/_collectionsmodule.c

```diff
@@ -1050,8 +1050,10 @@ deque_index(dequeobject *deque, PyObject *const *args, Py_ssize_t nargs)
         start = stop;
     assert(0 <= start && start <= stop && stop <= Py_SIZE(deque));
 
-    /* XXX Replace this loop with faster code from deque_item() */
-    for (i=0 ; i<start ; i++) {
+    for (i=0 ; i<start-BLOCKLEN ; i+=BLOCKLEN) {
```

Comment:
Please make the new code conforming PEP 7:
```c
for (i = 0; i < start - BLOCKLEN; i += BLOCKLEN) {
```
```c
for (; i < start; i++) {
```

Reply:
I think that these two loops can be replaced with a single loop:
```c
index = deque->leftindex + start;
for (; index >= BLOCKLEN; index -= BLOCKLEN) {
    b = b->rightlink;
}
n = stop - start;
```
(not tested)

Reply:
I find that harder to reason about.

Reply:
It took me a time to understand the code in this PR, but it looks cryptic again the next day. The code proposed by me is simpler and more efficient. I have tested it, it works. What a problem do you have with it?